File: livedata/src/main/java/cn/ljpc/livedata/MainActivity.java
package cn.ljpc.livedata;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import java.util.Timer;
import java.util.TimerTask;

public class MainActivity extends AppCompatActivity {

    private MyViewModel viewModel;

    private TextView timeView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        timeView = findViewById(R.id.time);

        viewModel = new ViewModelProvider(
                this, new ViewModelProvider.AndroidViewModelFactory(getApplication())).get(MyViewModel.class);
        viewModel.getTime().observe(this, time -> {
            timeView.setText(String.valueOf(time));
        });

        setTimer();
    }

    private void setTimer() {
        new Timer("我是一个定时器").schedule(new TimerTask() {
            @Override
            public void run() {
                viewModel.getTime().postValue(viewModel.getTime().getValue() + 1);
            }
        }, 0, 1000);
    }

    public void doAdd(View view){
        viewModel.getTime().postValue(viewModel.getTime().getValue() + 1);
    }
}

File: model/src/main/java/cn/ljpc/model/view/MainActivity.kt
package cn.ljpc.model.view

import android.os.Bundle
import android.util.Log
import android.widget.TextView
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import cn.ljpc.model.R
import cn.ljpc.model.viewmodel.ItemViewModel
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val model: ItemViewModel by viewModels()
        model.item.observe(this) {
            findViewById<TextView>(R.id.tx_show).text = "当前${it}次"
        }

        //对象表达式
        Timer().schedule(object : TimerTask() {
            override fun run() {
                model.item.postValue(model.item.value!! + 1)
                Log.d("lan", "model的值${model.item.value}")
            }
        }, 0, 1000)

    }
}